src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -512,7 +512,7 @@ public int GetHashCode(Symbol member)
                 }
 
                 if (_considerReturnType && member.GetMemberArity() == 0 && 
-                    (_typeComparison & TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) == 0) // If it is generic, then type argument might be in return type.
+                    (_typeComparison & TypeCompareKind.AllIgnoreOptions) == 0) // If it is generic, then type argument might be in return type.
```

Comment:
Isn't this ignoring new things besides nullability? Why the change?

Reply:
> Isn't this ignoring new things besides nullability? Why the change?

At this point I believe the change is necessary to ensure correctness because at the moment some implementations of GetHashCode on symbols actually sensitive to things that we are supposed to ignore. See [https://github.com/dotnet/roslyn/issues/30673](https://github.com/dotnet/roslyn/issues/30673). I'll add a note to that issue that once we ensure that all GetHashCode implementations do the right thing, this condition probably can be removed.


---
In reply to: [228374243](https://github.com/dotnet/roslyn/pull/30770#discussion_r228374243) [](ancestors = 228374243)